src/cartridges/int_adyen_overlay/cartridge/scripts/util/adyenHelper.js

```diff
@@ -664,6 +664,11 @@ var adyenHelperObj = {
       );
     }
 
+    const tokenAlreadyExists = paymentInstrument.getCreditCardToken();
+    if (!tokenAlreadyExists && result.additionalData && result.additionalData['recurring.recurringDetailReference']) {
```

Comment:
If there is a token, it will always be returned from Adyen, so we don't need this check